site/src/request_handlers/self_profile.rs

```diff
@@ -132,6 +136,19 @@ pub async fn handle_self_profile_processed_download(
         })
         .status(StatusCode::OK);
 
+    if output.filename.ends_with("json") && support_brotli {
```

Comment:
You could reuse `maybe_compressed_response` from the `server.rs` file. We just to check if the compression of the profiles is still good enough (i.e. that the resulting file is small enough) with the parameters configured in `server.rs`.